### DISCUSSION TITLE
Custom ViewModel naming - ConfigureViewModelLocator private in PrismAppBuilder

### DISCUSSION THREAD

**Discussion by jameslavery** [PrismLibrary/Prism#3002] [.NET MAUI] [answered]
I need to have a custom configuration for my ViewModel location.

In Forms we'd do it by overriding ConfigureViewModelLocator in app.xaml.cs, but with MAUI and Prism, I can't find a matching override. Looking at the source, ConfigureViewModelLocator is private in PrismAppBuilder.

Am I missing something, or is this not (yet) possible?

**Comment by dansiegel** [comment]
Can you expand on the question. Are you trying to configure the Type to Type mapping or resolving the ViewModel from once given the ViewModel type?

**Reply by jameslavery** [reply]
I need to configure a custom Page/ContentView to ViewModel naming convention. It's documented [here](https://prismlibrary.com/docs/viewmodel-locator.html), but this documentation refers to overriding a method in app.xaml.cs (presumably based on a Forms implementation), which isn't available in MAUI.

**Reply by dansiegel** [reply]
We can certainly look at making this easier but it's not that challenging as is... since you're already writing code in a static class you just want to do something like:

```cs
public static MauiApp CreateMauiApp() =>
    MauiApp.CreateBuilder()
        .UseMauiApp<App>()
        .UsePrism(prism => prism
            .ConfigureCustomViewModelLocatorOverride());

private static PrismAppBuilder ConfigureCustomViewModelLocatorOverride(this PrismAppBuilder)
{
    ViewModelLocationProvider.SetDefaultViewTypeToViewModelTypeResolver((viewType) =>
    {
        var viewName = viewType.FullName.Replace(".ViewModels.", ".CustomNamespace.");
        var viewAssemblyName = viewType.GetTypeInfo().Assembly.FullName;
        var viewModelName = $"{viewName}ViewModel, {viewAssemblyName}";
        return Type.GetType(viewModelName);
    });
}
```

**Reply by jameslavery** [reply, marked as answer]
Hi Dan - neat, hadn't thought of that, thank you.
